src/pytorch_lightning/loops/fit_loop.py

```diff
@@ -304,7 +304,10 @@ def on_advance_end(self) -> None:
         self.trainer._logger_connector.on_epoch_end()
 
         if self.epoch_loop._num_ready_batches_reached():
-            self.epoch_loop.update_lr_schedulers("epoch", update_plateau_schedulers=True)
+            # if we are restarting and the above condition holds, it's because we are reloading an epoch-end checkpoint.
```

Comment:
if we are reloading an epoch-end checkpoint, wouldn't it start from the next epoch and recompute the metrics?

Reply:
In fact, I am resuming from an epoch-end classic checkpoint (coming from ModelCheckpoint). What I observed, assuming the ended epoch was epoch 1, when resuming:
epoch 1 start > epoch 1 end > epoch 2 start > epoch 2 step > epoch 2 step > ...
It appears that if I am not setting strict=False to the ReduceLROnPlateau scheduler, it generates an error during epoch 1 end as no metric where computed for that epoch.

Reply:
@rohitgr7 Remember that depending on whether it was saved `on_train_epoch_end` vs `on_train_end`, we might need to re-run the end of the epoch that saved this checkpoint, given the fault-tolerance logic.

`num_ready_batches_reached` and `restarting` will only be True under this condition.